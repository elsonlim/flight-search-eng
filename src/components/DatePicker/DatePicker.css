.no-border .SingleDatePickerInput {
    border: none;
}

.no-border svg[role="presentation"] {
    display: none;
}

.CalendarMonth_table {
    margin-top: 10px;
}

.SingleDatePicker_picker {
    top: 45px !important;
}