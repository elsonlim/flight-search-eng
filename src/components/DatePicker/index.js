import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { SingleDatePicker } from 'react-dates';
import moment from 'moment';
import 'react-dates/initialize';
import 'react-dates/lib/css/_datepicker.css';
import './DatePicker.css';

class DatePicker extends Component {
    constructor(props) {
        super(props);
        this.state = {
          date: props.initialDate,
        };
    
        this.onDateChange = this.onDateChange.bind(this);
        this.onFocusChange = this.onFocusChange.bind(this);

        this.props.onDateChange(this.state.date)
      }
    
      onDateChange(date) {
        this.setState({ date });
        this.props.onDateChange(date)
      }
    
      onFocusChange({ focused }) {
        this.setState({ focused });
      }
    
      render() {
        const { focused, date } = this.state;
    
        return (
            <div className="no-border">
                <SingleDatePicker
                    id="date_input"
                    date={date}
                    focused={focused || false}
                    onDateChange={this.onDateChange}
                    onFocusChange={this.onFocusChange}
                    displayFormat={() => moment.localeData().longDateFormat('LL')}
                    hideKeyboardShortcutsPanel={true}
                    noBorder={false}
                    numberOfMonths={1}
                    block={true}
                />
            </div>
          
        );
      }
}

DatePicker.propTypes = {
  initialDate: PropTypes.object.isRequired,
  onDateChange: PropTypes.func.isRequired,
};

export default DatePicker;