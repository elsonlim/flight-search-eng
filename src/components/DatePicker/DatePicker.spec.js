import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import moment from 'moment';
import enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import DatePicker from './index';

enzyme.configure({ adapter: new Adapter() });

describe('DatePicker', () => {

    let props;
    let wrapper;
    
    beforeEach(() => {
        props = {
            initialDate: moment(new Date()),
            onDateChange: jest.fn(),
        }
        wrapper = shallow(<DatePicker {...props} />);
    });

    it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<DatePicker {...props} />, div);
		ReactDOM.unmountComponentAtNode(div);
    });
    
    describe('snapshot test', () => {
		test('show airline logo', () => {
			const tree = renderer.create(<DatePicker {...props} />).toJSON();
			expect(tree).toMatchSnapshot();
		});
    });
    
    describe('onDateChange', () => {
        test('update the date value in state', () => {
            const dateObj = moment(new Date());
            wrapper.instance().onDateChange(dateObj);

            const date = wrapper.state().date;
            expect(date).toEqual(dateObj);
        });

        test('calls props onDateChange with date', () => {
            const dateObj = moment(new Date());
            props.onDateChange.mockClear();
            
            wrapper.instance().onDateChange(dateObj);
            
            expect(props.onDateChange).toHaveBeenCalledTimes(1);
        });
    });

    describe('onFocusChange', () => {
        test('update the focus value in state', () => {
            wrapper.instance().onFocusChange({
                focused: true
            });

            const focused = wrapper.state().focused;
            expect(focused).toEqual(true);
        });
    });
});
