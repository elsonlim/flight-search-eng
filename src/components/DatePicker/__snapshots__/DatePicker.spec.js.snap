// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker snapshot test show airline logo 1`] = `
<div
  className="no-border"
>
  <div
    className="SingleDatePicker SingleDatePicker_1 SingleDatePicker__block SingleDatePicker__block_2"
  >
    <div>
      <div
        className="SingleDatePickerInput SingleDatePickerInput_1 SingleDatePickerInput__withBorder SingleDatePickerInput__withBorder_2 SingleDatePickerInput__block SingleDatePickerInput__block_3"
      >
        <div
          className="DateInput DateInput_1 DateInput__block DateInput__block_2"
        >
          <input
            aria-describedby="DateInput__screen-reader-message-date_input"
            aria-label="Date"
            autoComplete="off"
            className="DateInput_input DateInput_input_1"
            disabled={false}
            id="date_input"
            name="date_input"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Date"
            readOnly={false}
            required={false}
            type="text"
            value="May 20, 2018"
          />
          <p
            className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
            id="DateInput__screen-reader-message-date_input"
          >
            Press the down arrow key to interact with the calendar and
              select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
